### PR TITLE
feat: update referral cash instant

### PR DESCRIPTION
**Pull request by dhruv1294**

### CODE REVIEW

File: app/src/main/java/org/pragyan/dalal18/ui/VerifyPhoneActivity.kt [hunk 1]
```diff
@@ -195,12 +209,97 @@ class VerifyPhoneActivity : AppCompatActivity(), ConnectionUtils.SmsVerification
...
+        }
+
+    private fun startLoginProcess(email: String?, password: String?) {
+
+        if (email != null && password != null && email != "") {
+            loginAsynchronously(email, password)
+        } else {
+            startLoginActivity()
+        }
+    }
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/ui/VerifyPhoneActivity.kt:287]: Eliminate this function, and use this body phoneVerificationSuccessful() function

File: app/src/main/java/org/pragyan/dalal18/ui/VerifyPhoneActivity.kt [hunk 2]
```diff
@@ -195,12 +209,97 @@ class VerifyPhoneActivity : AppCompatActivity(), ConnectionUtils.SmsVerification
     }
 
     override fun phoneVerificationSuccessful() {
-        val intent = Intent(this, MainActivity::class.java)
-        intent.putExtras(this.intent)
-        startActivity(intent)
+
+        startLoginProcess(preferences.getString(SplashActivity.EMAIL_KEY, null), preferences.getString(SplashActivity.PASSWORD_KEY, null))
+        //startActivity(intent)
+        //finish()
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/ui/VerifyPhoneActivity.kt:215]: Remove the comments

File: app/src/main/java/org/pragyan/dalal18/ui/VerifyPhoneActivity.kt [hunk 3]
```diff
@@ -195,12 +209,97 @@ class VerifyPhoneActivity : AppCompatActivity(), ConnectionUtils.SmsVerification
...
+                                    putExtra(MainActivity.TOTAL_WORTH_KEY, loginResponse.user.total)
+                                    putExtra(MainActivity.RESERVED_CASH_KEY, loginResponse.user.reservedCash)
+                                }
+                                startActivity(intent)
+                                finish()
+                            }
+                        }
+                    }
+                         else {
+                            uiThread { startLoginActivity() }
```
Comment (avikumar15) [inline app/src/main/java/org/pragyan/dalal18/ui/VerifyPhoneActivity.kt:248]: Follow the intendation here

**Review by avikumar15 — APPROVED** [review]

File: app/src/main/java/org/pragyan/dalal18/ui/VerifyPhoneActivity.kt [hunk 4]
```diff
@@ -195,12 +209,93 @@ class VerifyPhoneActivity : AppCompatActivity(), ConnectionUtils.SmsVerification
...
+    private fun getStubMetadata(sessionId: String): Metadata {
+        val metadata = Metadata()
+        val metadataKey = Metadata.Key.of("sessionid", Metadata.ASCII_STRING_MARSHALLER)
+        metadata.put(metadataKey, sessionId)
+        return metadata
+    }
+
+
     override fun onNetworkDownError(message: String) {
         toast(message)
```
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/ui/VerifyPhoneActivity.kt:300]: should have used Toast.make(...) instead of **anko** :(

**Review by ranaaditya — APPROVED** [review]

**Pull request merged**